src/hooks/useResizeObserver.ts

```diff
@@ -9,26 +9,29 @@ import { useCallback, useEffect, useRef, useState } from 'react';
  * - height - Height of the container
  */
 export const useResizeObserver = () => {
-  const resizeObserver = useRef<ResizeObserver>();
+  const [node, setNode] = useState<HTMLElement | null>(null);
   const [width, setWidth] = useState<number>();
   const [height, setHeight] = useState<number>();
 
   const ref = useCallback((node: HTMLElement | null) => {
+    setNode(node);
+  }, []);
+
+  useLayoutEffect(() => {
     if (!node) return;
-    resizeObserver.current = new ResizeObserver(() => {
+    const observer = new ResizeObserver((entries) => {
```

Comment:
In React 18, due to changes in StrictMode and concurrent rendering, the timing and order of callback ref invocations may differ from before, causing the ResizeObserver not to trigger as expected and resulting in the state’s height remaining undefined.

Reply:
@patrikk0123 , are you ok with related changes?